lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm

```diff
@@ -17,17 +17,44 @@
 package OpenQA::WebAPI::Plugin::ObsRsync::Controller::Folders;
 use Mojo::Base 'Mojolicious::Controller';
 use Mojo::File;
+use POSIX 'strftime';
+use Data::Dumper;
 
 sub index {
     my $self   = shift;
-    my $folder = $self->param('folder');
     my $helper = $self->obs_rsync;
-    return undef if $helper->check_and_render_error($folder);
-    my $files
+    my %folders_info;
+    my $folders
       = Mojo::File->new($helper->home)->list({dir => 1})->grep(sub { -d $_ })->map('basename')->grep(qr/^(?!t$|sle$)/)
       ->to_array;
+    for my $folder (@$folders) {
+        $folders_info{$folder} = {
+            run_last     => $helper->get_run_last_info($folder),
+            dirty_status => $helper->get_dirty_status($folder)};
+    }
+
+    my $running_jobs = $self->app->minion->backend->list_jobs(
+        0,
+        4 * $helper->queue_limit,
+        {tasks => ['obs_rsync_run'], states => ['active', 'inactive']});
+
+    for my $job (@{$running_jobs->{jobs}}) {
+        my $args = $job->{args};
+        $args = $args->[0] if (ref $args eq 'ARRAY' && scalar(@$args) == 1);
 
-    $self->render('ObsRsync_index', folders => $files);
+        if (ref $args eq 'HASH' && scalar(%$args) == 1 && $args->{project}) {
```

Comment:
I'd rather structure the code using `next unless ...` to avoid deeply nested code.